src/affine.py

```diff
@@ -159,7 +153,9 @@ def _check_i(self, attribute, value):
             raise ValueError("i must be equal to 1.0")
 
     @classmethod
-    def from_gdal(cls, c: float, a: float, b: float, f: float, d: float, e: float):
+    def from_gdal(
+        cls, c: float, a: float, b: float, f: float, d: float, e: float
+    ) -> "Affine":
```

Comment:
mypy was fine with `Affine`, but ruff insists on quotes 🤷

Reply:
I'm not a typing pro, but after a 20 second read of [PEP 649](https://peps.python.org/pep-0649/) it is potentially related. Add `from __future__ import annotations` to the top which should make ruff happy.